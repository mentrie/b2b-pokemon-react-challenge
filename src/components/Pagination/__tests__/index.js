import React from 'react';
import Pagination from '../index';

const props = {
  current: 0,
  total: 5,
  onPrev: jest.fn(),
  onNext: jest.fn()
};

test('renders component properly', () => {
  const component = shallow(<Pagination {...props} />);
  expect(component).toMatchSnapshot();
});

test('should not render if there are no pages', () => {
  const newProps = { ...props, total: 0 };
  const component = shallow(<Pagination {...newProps} />);

  expect(component.find('.pagination').length).toBe(0);
});

test('should not render next button if current is greater than or requal total', () => {
  const newProps = { ...props, current: 5 };
  const component = shallow(<Pagination {...newProps} />);

  expect(component.find('.pagination__next').length).toBe(0);
});

test('should not render prev button if currentPage is less than one', () => {
  const newProps = { ...props, current: 0 };
  const component = shallow(<Pagination {...newProps} />);

  expect(component.find('.pagination__prev').length).toBe(0);
});
