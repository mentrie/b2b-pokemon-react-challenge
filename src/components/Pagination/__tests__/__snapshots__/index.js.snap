// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component properly 1`] = `
<div
  className="pagination"
>
  <div
    className="pagination__current"
  >
    0
  </div>
  <span
    className="pagination__text"
  >
     of 
    5
  </span>
  <div
    className="pagination__next"
    onClick={[Function]}
  >
    Next
  </div>
</div>
`;
